src/commonMain/kotlin/exchange.dydx.abacus/state/manager/StateManagerAdaptor.kt

```diff
@@ -1750,6 +1757,121 @@ open class StateManagerAdaptor(
+        val goodTilTimeInSeconds = (duration / 1.seconds).toInt()
+
+        val marketInfo = marketInfo(marketId)
+        val currentHeight = calculateCurrentHeight()
+
+        return HumanReadablePlaceOrderPayload(
+            subaccountNumber,
+            marketId,
+            clientId,
+            type,
+            side,
+            price,
+            triggerPrice,
+            size,
+            reduceOnly,
+            postOnly,
+            timeInForce,
+            execution,
+            goodTilTimeInSeconds,
+            marketInfo,
+            currentHeight,
+        )
+    }
+
+    private fun isTriggerOrderEqualToExistingOrder(triggerOrder: TriggerOrder, existingOrder: SubaccountOrder, size: Double): Boolean {
+        val limitPriceCheck = when (triggerOrder.type) {
+            OrderType.stopLimit, OrderType.takeProfitLimit -> triggerOrder.price?.limitPrice == existingOrder.price
+            else -> true
+        }
+        return size == existingOrder.size &&
+            triggerOrder.type == existingOrder.type &&
+            triggerOrder.side == existingOrder.side &&
+            triggerOrder.price?.triggerPrice == existingOrder.triggerPrice &&
+            limitPriceCheck
+    }
+
+    @Throws(Exception::class)
+    fun triggerOrdersPayload(): HumanReadableTriggerOrdersPayload {
+        val placeOrderPayloads = mutableListOf<HumanReadablePlaceOrderPayload>()
+        val cancelOrderPayloads = mutableListOf<HumanReadableCancelOrderPayload>()
+        val triggerOrders = stateMachine.state?.input?.triggerOrders
+
+        val subaccountNumber = connectedSubaccountNumber ?: throw Exception("subaccountNumber is null")
+        val subaccount = stateMachine.state?.subaccount(subaccountNumber) ?: throw Exception("subaccount is null")
+
+        val marketId = triggerOrders?.marketId ?: throw Exception("marketId is null")
+        val size = triggerOrders.size ?: throw Exception("size is null")
+
+        fun updateTriggerOrder(triggerOrder: TriggerOrder) {
+            // Cases
+            // 1. Existing order -> update
+            // 2. Existing order -> nothing should be done
+            // 3. Existing order -> should delete
+            // 4. No existing order -> create a new one
+            // 5. No existing order -> nothing should be done
+
+            if (triggerOrder.orderId != null) {
+                val existingOrder = subaccount.orders?.firstOrNull { it.id == triggerOrder.orderId }
+                    ?: throw Exception("order is null")
+                if (triggerOrder.price?.triggerPrice != null) {
+                    if (!isTriggerOrderEqualToExistingOrder(triggerOrder, existingOrder, size)) {
+                        // (1) Existing order -> update
+                        cancelOrderPayloads.add(cancelOrderPayload(triggerOrder.orderId))
+                        placeOrderPayloads.add(triggerOrderPayload(triggerOrder, subaccountNumber, marketId, size))
```

Comment:
thoughts on using two vars `shouldCancelOrder` and `shouldPlaceOrder`, so these two lines don't have to be duplicated / potentially easier to read since it's just the same order?

Reply:
hm I tried this out and felt like it was actually harder to parse out what was happening - so leaning towards keeping it like this if that's alright with yoU!